### PR TITLE
[SofaBoundaryCondition] Sanitize

### PR DESCRIPTION
**Pull request by alxbilger**
- Hiding local declaration 
- Parameter names do not match 
- Uninitialized class members 
- Lambda capture is never used 
- Local variable can be made const 
- Result of a postfix operator is discarded 
- add explicit specifier to make constructor non-converting 
- Member function can be made const 
- Variable can be moved to if-statement 
- Declaration and assignment can be joined 
- Return by const value

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Comment by fredroy** [comment]
[ci-build][with-all-tests]

File: modules/SofaBoundaryCondition/src/SofaBoundaryCondition/FixedRotationConstraint.inl [hunk 1]
```diff
@@ -87,9 +87,9 @@ void FixedRotationConstraint<DataTypes>::projectPosition(const core::MechanicalP
     for (unsigned int i = 0; i < x.size(); ++i)
     {
         // Current orientations
-        sofa::type::Quat<SReal> Q = x[i].getOrientation();
+        const sofa::type::Quat<SReal> Q = x[i].getOrientation();
```
Comment (fredroy) [inline modules/SofaBoundaryCondition/src/SofaBoundaryCondition/FixedRotationConstraint.inl]:
```suggestion
        const sofa::type::Quat<SReal>& Q = x[i].getOrientation();
```
In this case, you might be able to use a ref

File: modules/SofaBoundaryCondition/src/SofaBoundaryCondition/FixedRotationConstraint.inl [hunk 2]
```diff
@@ -87,9 +87,9 @@ void FixedRotationConstraint<DataTypes>::projectPosition(const core::MechanicalP
...
         // Previous orientations
-        sofa::type::Quat<SReal> Q_prev = previousOrientation[i];
+        const sofa::type::Quat<SReal> Q_prev = previousOrientation[i];
```
Comment (fredroy) [inline modules/SofaBoundaryCondition/src/SofaBoundaryCondition/FixedRotationConstraint.inl]:
```suggestion
        const sofa::type::Quat<SReal>& Q_prev = previousOrientation[i];
```
same (could not do a multi-line suggestion 😕)

**Review by fredroy — CHANGES_REQUESTED** [review]
Overall a nice cleaning PR 👍

File: modules/SofaBoundaryCondition/src/SofaBoundaryCondition/FixedRotationConstraint.inl (hunk 2 above)
Comment (alxbilger) [reply modules/SofaBoundaryCondition/src/SofaBoundaryCondition/FixedRotationConstraint.inl]: Actually, you can select multiple lines before making the suggestion

**Pull request merged**